feat: ability to supply deadline Python wheel

godobyte opened the pull request:
### What was the problem/requirement? (What/Why)
There is no ability to supply a custom developer version of `deadline-cloud` and have it deployed with the worker instances launched by `deadline-cloud-test-fixtures`

### What was the solution? (How)
Introduce a new `DEADLINE_WHL_PATH` environment variable that contains a glob pattern resolving to a local path to a .whl file for `dealine-cloud` on the test machine. This works the same as the existing WORKER_AGENT_WHL_PATH environment variable for supplying a custom developer version of the worker agent.

### What is the impact of this change?
Consumers of the library can orchestrate tests of workers using a custom supplied deadline Python library.

### How was this change tested?

1. Set the DEADLINE_WHL_PATH to the path of a developer-built .whl file
2. Ran the tests
3. Inspected the test logs 

```
[2025-01-13 15:53:13,468] [test/e2e/test_job_submissions.py::TestJobSubmission::test_success] Using WORKER_AGENT_WHL_PATH whl file: /deadline-wa/deadline-cloud-worker-agent/dist/deadline_cloud_worker_agent-0.0.post1.dev370-py3-none-any.whl
[2025-01-13 15:53:13,468] [test/e2e/test_job_submissions.py::TestJobSubmission::test_success] The WORKER_AGENT_WHL_PATH whl file will be copied to /tmp/deadline_cloud_worker_agent-0.0.post1.dev370-py3-none-any.whl on the Worker environment
[2025-01-13 15:53:13,468] [test/e2e/test_job_submissions.py::TestJobSubmission::test_success] Using DEADLINE_WHL_PATH whl file: /deadline-wa/deadline-cloud/dist/deadline-0.0.post381+ge3f849c-py3-none-any.whl
[2025-01-13 15:53:13,469] [test/e2e/test_job_submissions.py::TestJobSubmission::test_success] The DEADLINE_WHL_PATH whl file will be copied to /tmp/deadline-0.0.post381+ge3f849c-py3-none-any.whl on the Worker environment
```

### Was this change documented?
Yes, the new environment variable was added to `src/deadline_test_fixtures/example_config.sh` where the other environment variables are documented.

### Is this a breaking change?
No

----

*By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.*

godobyte committed:
feat: ability to supply deadline Python wheel

Signed-off-by: Godot Bian <13778003+godobyte@users.noreply.github.com>

jusiskin commented:
The PR description says:

>  1.  Modified the worker agent e2e test suite to extend the worker_config fixture from deadline-cloud-test-fixtures instead of reimplementing it

But I believe this is a copy/paste from #165, right? This change has already been merged in aws-deadline/deadline-cloud-worker-agent#492:

jusiskin commented on src/deadline_test_fixtures/fixtures.py:
```diff
@@ -401,6 +401,35 @@ def deletable(resource: T) -> Generator[T, None, None]:
+        raise NotImplementedError(f"Unsupported operating system: {operating_system.name}")
+
+    LOG.info(f"The {env_var_name} whl file will be copied to {dest_path} on the Worker environment")
+    return (resolved_whl_path, dest_path)
```
The first element in this tuple seems to never be used by callers. Do we need it?

godobyte replied:
The first element is needed and is in use. `file_mappings` below takes a list of tuples, i.e. `file_mappings.append((resolved_whl_path, dest_path))`.

jusiskin replied:
Oops, I missed that. Thanks for the correction

jusiskin reviewed (approved):
Looks good. One small nitpick but it is not merge-blocking.

godobyte commented:
> The PR description says:
> 
> > 1. Modified the worker agent e2e test suite to extend the worker_config fixture from deadline-cloud-test-fixtures instead of reimplementing it
> 
> But I believe this is a copy/paste from #165, right? This change has already been merged in [aws-deadline/deadline-cloud-worker-agent#492](https://github.com/aws-deadline/deadline-cloud-worker-agent/pull/492):

Good catch, updated to remove that from description.
